app/src/main/java/com/door43/translationstudio/newui/translate/TargetTranslationActivity.java

```diff
@@ -300,6 +349,240 @@ public boolean onMenuItemClick(MenuItem item) {
+                ArrayAdapter<String> typesAdapter = new ArrayAdapter<String>(this, android.R.layout.simple_spinner_item, types);
+                typesAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
+                type.setAdapter(typesAdapter);
+
+                type.setOnItemSelectedListener(this);
+            }
+        }
+    }
+
+    /**
+     * sets the visible state of the search spinner
+     * @param displayed
+     */
+    private void setSearchSpinner(boolean displayed) {
+        if(mSearchingSpinner != null) {
+            mSearchingSpinner.setVisibility(displayed ?  View.VISIBLE : View.GONE);
+        }
+    }
+
+    /**
+     * called if search type is changed
+     * @param parent
+     * @param view
+     * @param pos
+     * @param id
+     */
+    public void onItemSelected(AdapterView<?> parent, View view,
+                               int pos, long id) {
+        setSearchFilter();
+    }
+
+    /**
+     * called if no search type is selected
+     * @param parent
+     */
+    public void onNothingSelected(AdapterView<?> parent) {
+        // do nothing
+    }
+
+    /**
+     * see if translation search is selected
+     */
+    private boolean isTranslationSearchSelected() {
+        int position = getSearchTypeSelection();
+        return position == 1;
+    }
+
+    /**
+     * get the type of search (position)
+     */
+    private int getSearchTypeSelection() {
+        LinearLayout searchPane = (LinearLayout) findViewById(R.id.search_pane);
+        if(searchPane != null) {
+
+            Spinner type = (Spinner) searchPane.findViewById(R.id.search_type);
+            if(type != null) {
+                int pos = type.getSelectedItemPosition();
+                if(pos >= 0) {
+                    return pos;
+                }
+            }
+        }
+
+        return 0;
+    }
+
+    /**
+     * get search text in search bar
+     */
+    private String getSearchText() {
+        String text = null;
+
+        LinearLayout searchPane = (LinearLayout) findViewById(R.id.search_pane);
+        if(searchPane != null) {
+
+            EditText edit = (EditText) searchPane.findViewById(R.id.search_text);
+            if(edit != null) {
+                text = edit.getText().toString();
+            }
+        }
+        return text;
+    }
+
+    /**
+     * set search text in search bar
+     */
+    private void setSearchText(String text) {
+
+        if(text == null) {
+            text = "";
+        }
+
+        LinearLayout searchPane = (LinearLayout) findViewById(R.id.search_pane);
+        if(searchPane != null) {
+
+            EditText edit = (EditText) searchPane.findViewById(R.id.search_text);
+            if(edit != null) {
+                edit.setText(text);
+            }
+        }
+    }
+
+    /**
+     * do search with search string in edit control
+     */
+    public void setSearchFilter() {
```

Comment:
This is used only once and is just one line. It does not need to be in a function.